cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AbstractNotificationSubscriberService.java

```diff
@@ -107,97 +105,148 @@ protected void shutDown() {
   /**
    * Thread that subscribes to TMS notifications on a specified topic and fetches notifications, retrying if necessary
    */
-  protected abstract class NotificationSubscriberThread implements Runnable {
+  protected abstract class AbstractSubscriberRunnable implements Runnable {
+
     private final String topic;
+    private final boolean transactionalFetch;
+    private final long emptyFetchDelayMillis;
     private final RetryStrategy retryStrategy;
+    private final MetricsContext metricsContext;
     private int failureCount;
     private String messageId;
 
-    protected NotificationSubscriberThread(String topic) {
+    protected AbstractSubscriberRunnable(String name, String topic, boolean transactionalFetch) {
       this.topic = topic;
-      // TODO: [CDAP-11370] Need to be configured in cdap-default.xml. Retry with delay ranging from 0.1s to 30s
-      retryStrategy =
-        co.cask.cdap.common.service.RetryStrategies.exponentialDelay(100, 30000, TimeUnit.MILLISECONDS);
+      this.transactionalFetch = transactionalFetch;
+      this.emptyFetchDelayMillis = cConf.getLong(Constants.Scheduler.EVENT_POLL_DELAY_MILLIS);
+      this.retryStrategy = RetryStrategies.fromConfiguration(cConf, "system.notification.");
```

Comment:
I think `ProgramNotificationSubscriberService.java` should have different retry strategies from `ScheduleNotificationSubscriberService.java`, even different topics in `ScheduleNotificationSubscriberService.java` can have different retry strategies.